internal/memory/git_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Lekko Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/storage"
+	gitmemory "github.com/go-git/go-git/v5/storage/memory"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+func TestGitStore(t *testing.T) {
+	ctx := context.Background()
+	storer, fs := setupFS(t)
+
+	gs, err := newGitStore(ctx, "", "lekkodev", "testrepo", storer, fs, nil, 10, false)
+	require.NoError(t, err)
+	assert.NotNil(t, gs)
+	assert.NotContains(t, gs.store.getCommitSha(), "dirty", "working directory is clean")
+
+	bv := wrapperspb.BoolValue{}
+	require.NoError(t, gs.Evaluate("example", "default", nil, &bv))
+	assert.False(t, bv.Value)
+
+	sv := wrapperspb.StringValue{}
+	require.NoError(t, gs.Evaluate("string", "test-namespace", nil, &sv))
+	assert.Equal(t, sv.Value, "")
+
+	require.Error(t, gs.Evaluate("not-a-key", "not-a-ns", nil, &bv))
+
+	require.NoError(t, gs.Close(ctx))
+}
+
+func setupFS(t *testing.T) (storage.Storer, billy.Filesystem) {
+	storer := gitmemory.NewStorage()
+	fs := memfs.New()
```

Comment:
unfortunately it doesn't. the watching library doesn't work with virtual filesystems, it assumes native `os` filesystem. to test it would require an integration test. i plan to add one in a later PR though, we can have it run during CI on github actions.